### PR TITLE
Bounty claim profitability

### PR DESCRIPTION
**Pull request by Bagaric**
Adds a check if the Bounty set by the scheduler is high enough to be profitable to be claimed.

`PROFITABILITY_INDEX` determined through experimentation by @kosecki123.

### CODE REVIEW

File: client/routing.js [hunk 1]
```diff
@@ -33,8 +38,10 @@ const isProfitableToClaim = async (conf, txRequest, gasToClaim) => {
   const currentGasPrice = new BigNumber(await Util.getGasPrice(web3))
   const gasCostToClaim = currentGasPrice.times(gasToClaim)
 
-  if (gasCostToClaim.greaterThan(paymentWhenClaimed)) {
-    conf.logger.debug(`[${txRequest.address}] Not profitable to claim. gasCostToClaim: ${gasCostToClaim} | paymentWhenClaimed: ${paymentWhenClaimed}`)
+  const bountyProfitable = txRequest.bounty > (PROFITABILITY_INDEX * currentGasPrice)
```
Comment (Kuzirashi) [inline client/routing.js]: Isn't a `txRequest.bounty` a BigNumber? Please check whether you should use here `>` or `.greaterThan()`.

File: client/routing.js [hunk 2]
```diff
@@ -10,6 +10,11 @@ const STATE = {
   DONE: 4,
 }
 
+// Based on experimentation, we've found that the bounty
+// needs to be at least `profitabilityIndex * currentGasPrice`
+// to be profitable.
+const PROFITABILITY_INDEX = 100000;
```
Comment (lsaether) [inline client/routing.js]: I don't think I understand the difference between the PROFITABILITY_INDEX and roughly the gas it costs to send the claim transaction. I think it's redundant.

**Comment by Bagaric** [comment]
@kosecki123 We should think how to quantify this probability index and display it to the user. Suggestion: 100000 is the default (minimum) and the user can choose between 1x to 5x of the default.
What do you think?

File: client/routing.js (hunk 2 above)
Comment (Bagaric) [reply client/routing.js]: I think here we're trying to stop TimeNodes from claiming transactions that will not bring enough bounty to be profitable after execution. @kosecki123 can probably explain it a bit better.

**Comment by Bagaric** [comment]
Orphaned. Closing for now.

**Pull request closed without merging**